jiant/tasks/qa.py

```diff
@@ -273,10 +273,12 @@ def tokenize_preserve_placeholder(sent):
             """ Tokenize questions while preserving @placeholder token """
             sent_parts = sent.split("@placeholder")
             assert len(sent_parts) == 2
-            sent_parts = [
-                tokenize_and_truncate(self.tokenizer_name, s, self.max_seq_len) for s in sent_parts
-            ]
-            return sent_parts[0] + ["@placeholder"] + sent_parts[1]
+            placeholder_loc = len(
+                tokenize_and_truncate(self.tokenizer_name, sent_parts[0], self.max_seq_len)
+            )
+            sent_tok = tokenize_and_truncate(self.tokenizer_name, sent, self.max_seq_len)
```

Comment:
This seems a little sketchy - prefer something like `return sent_tok[:i] + ["@placeholder"] + sent_tok[i:]` than mutating with insert?

Also, should this be truncated to `self.max_seq_len - ` to account for the placeholder?

Reply:
Thanks.